File: padroesComportamentais/padrao-de-projeto-iterator/java/src/main/java/Loja.java
public class Loja {

    public Float contaValorTotalCarrinho(Carrinho carrinho) {
        float valorTotal = 0f;
        for(Produto produto : carrinho) {
            valorTotal += produto.getValor();
        }
        return valorTotal;
    }

    public int contaTotalProdutosCarrinho(Carrinho carrinho) {
        int quantidade = 0;
        for(Produto produto : carrinho) {
            quantidade++;
        }
        return quantidade;
    }
}

File: padroesComportamentais/padrao-de-projeto-iterator/php/src/Loja.php
<?php

namespace Iterator\Php;

class Loja
{
    public function contaValorTotalCarrinho(Carrinho $carrinho): float
    {
        $quantidade = 0.0;
        /** @var Produto $produto */
        foreach($carrinho as $produto) {
            $quantidade += $produto->getValor();
        }
        return $quantidade;
    }

    public function contaTotalProdutosCarrinho(Carrinho $carrinho): int
    {
        $quantidade = 0;
        /** @var Produto $produto */
        foreach($carrinho as $produto) {
            $quantidade++;
        }
        return $quantidade;
    }
}